src/TrackingOptOut.php

```diff
@@ -9,60 +9,84 @@
 
 class TrackingOptOut {
 
-  /**
-   * Default Google Analytics opt out script.
-   *
-   * @var string
-   */
-  private static $opt_out_script = <<<CODE
-<script>
-// Disable tracking if the opt-out cookie exists.
-for (var i = 0; i < window.gaIdList.length; i++) {
-  var disableStr = 'ga-disable-' + window.gaIdList[i];
-  if (document.cookie.indexOf(disableStr + '=true') > -1) {
-    window[disableStr] = true;
-  }
-}
-/**
- * Sets cookies to indicate Google Analytics tracking should be disabled.
- */
-function gaOptout() {
-  var disableStr;
-  for (var i = 0; i < window.gaIdList.length; i++) {
-    disableStr = 'ga-disable-' + window.gaIdList[i];
-    document.cookie = disableStr + '=true; expires=Thu, 31 Dec 2099 23:59:59 UTC; path=/';
-    window[disableStr] = true;
-  }
-}
-</script>
-
-CODE;
-
   /**
    * @implements admin_init
    */
   public static function admin_init() {
-    // Register opt-out related settings.
-    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_ids');
-    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_script');
-    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_active');
-
+    static::addSettingsFields();
     add_action('admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts');
   }
 
   /**
    * @implements init
    */
   public static function init() {
-    // Injects the Google Analytics IDs and the opt-out script in the page head
-    // section before any tracking script.
-    add_action('wp_head', __CLASS__ . '::wp_head', 5);
 
     // Register the shortcode to call the opt-out cookie creation script.
     add_shortcode('ga-optout-link', __CLASS__ . '::registerOptOutShortcode');
 
     // Enqueue frontend scripts.
-    add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+    if (get_option('core_standards_optout_active')) {
+      add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+      // Injects the Google Analytics IDs and the opt-out script in the page head
+      // section before any tracking script.
+      add_action('wp_head', __CLASS__ . '::wp_head', 5);
+      add_action('wp_enqueue_scripts', __CLASS__ . '::enqueueScriptsFirstHead', 1);
+    }
+  }
+
+  /**
+   * Adds Opt-Out section settings fields.
+   */
+  public static function addSettingsFields() {
+    add_settings_section(
+      Plugin::PREFIX . '-optout',
+      __('WordPress Core Standards Settings', Plugin::L10N),
```

Comment:
1. A separate section for a single setting is a bit overkill.  New sections should only be added wisely.

1. If/when adding new sections, they should follow the UI guidelines of WordPress; i.e., their labels/titles should focus on functionality/perspective instead of a technical implementation detail (a plugin name).

1. The Reading settings page still appears to be a much better fit than the general settings page: https://gartenmoebelcompany.de/wp-admin/options-reading.php – for example, below the "Search engine visibility" setting.